mlflow/utils/rest_utils.py

```diff
@@ -169,6 +171,48 @@ def call_endpoint(host_creds, endpoint, method, json_body, response_proto):
     return response_proto
 
 
+# Response codes that generally indicate transient network failures and merit client retries,
+# based on guidance from cloud service providers
+# (https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#general-rest-and-retry-guidelines)
+TRANSIENT_FAILURE_RESPONSE_CODES = [
+    408, # Request Timeout
+    429, # Too Many Requests
+    500, # Internal Server Error
+    502, # Bad Gateway
+    503, # Service Unavailable
+    504, # Gateway Timeout
+]
+
+
+def http_put(*args, **kwargs):
```

Comment:
Can we further distinguish this method from the preexisting `http_request` function in this module by calling this `cloud_storage_http_put` and adds some docs indicating that this implements retries for an HTTP put operation in accordance with cloud vendor guidelines for storage APIs?

Reply:
In conjunction with https://github.com/mlflow/mlflow/pull/4044/files#r569026315, it might be good to define a single `cloud_storage_http_request` method that can be used for PUTs and GETs based on a caller-specified `method` argument.